src/functional.jl

```diff
@@ -0,0 +1,111 @@
+module FunctionalNullableOperations
+
+# extends Base with operations that treat nullables as collections
+
+importall Base
+import Base: promote_op, LinearFast
+
+# conceptually, nullable types can be considered infinite-order tensor powers of
+# finite-dimensional vector spaces — we attempt to support most operations on
+# arrays except the linear algebra related ones; the infinite-dimensional nature
+# makes subtyping AbstractArray a little dangerous, which explains why
+# functionality is reimplemented instead of subtyping AbstractArray
+
+# size   – not implemented since an infinite-dimensional tuple would be strange
```

Comment:
Why not consider `Nullable` as a 0-dimensional array instead? It's really similar. Then its length could be the empty tuple:
```julia
julia> x=Array(Int)
0-dimensional Array{Int64,0}:
139986151483248

julia> size(x)
()

```